src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1218,6 +1231,10 @@ CHIP_ERROR CASESession::OnMessageReceived(ExchangeContext * ec, const PacketHead
                                           const PayloadHeader & payloadHeader, System::PacketBufferHandle && msg)
 {
     CHIP_ERROR err = ValidateReceivedMessage(ec, packetHeader, payloadHeader, msg);
+    if (err == CHIP_ERROR_DUPLICATE_MESSAGE_RECEIVED)
+    {
+        ExitNow(err = CHIP_NO_ERROR);
+    }
```

Comment:
Defensive coding wise, why would we remove this?

Reply:
Because if this passed through the underlying layers, there is a big error happening. We should error-out since the underlying layer should have not given it to us. Therefore the correct defense is not to ignore, but to error.